### PR TITLE
e2e: make demo pod logged only when failed running

### PR DESCRIPTION
**Pull request by hj-johannes-lee**
fixes: #1252

### CODE REVIEW

File: test/e2e/gpu/gpu.go
Comment (tkatila) [inline test/e2e/gpu/gpu.go:101]: What's the timeout after the change?

File: test/e2e/qat/qatplugin_dpdk.go
Comment (tkatila) [inline test/e2e/qat/qatplugin_dpdk.go:139]: Timeout value?

File: test/e2e/gpu/gpu.go
Comment (tkatila) [inline test/e2e/gpu/gpu.go:103]: While at it, you could add gpu log output as well.

File: test/e2e/gpu/gpu.go
Comment (hj-johannes-lee) [reply test/e2e/gpu/gpu.go]: sorry, I didn't know there is 'WaitForPodSuccessInNamespaceTimeout'. I used and put the same timeout as before.!

File: test/e2e/gpu/gpu.go
Comment (hj-johannes-lee) [reply test/e2e/gpu/gpu.go:103]: fixed!

**Comment by hj-johannes-lee** [comment]
There are some parts that we did not put log printing after running a demo pod.
https://github.com/intel/intel-device-plugins-for-kubernetes/blob/c7feb2db66f7df03ab649da29f298ad852178459/test/e2e/fpga/fpga.go#L112
https://github.com/intel/intel-device-plugins-for-kubernetes/blob/main/test/e2e/qat/qatplugin_dpdk.go#L172

Then, do we add it in this PR?

**Comment by mythi** [comment]
> Then, do we add it in this PR? Or was there any reason that we do not need to check the logs from those?

No reason. In the past each e2e module was worked on independently by different individuals and that made the implementations to differ. I hope your work can make them to be more consistent.

If this PR plans to fix #1252, let's do that for all of the devices then and so that they work the same. Note that the ask for #1252 to not log output when there are no errors and only log when the sample app fails.

**Comment by hj-johannes-lee** [comment]
> No reason. In the past each e2e module was worked on independently by different individuals and that made the implementations to differ. I hope your work can make them to be more consistent.

I see. I felt that it would be good to have a common function for pod logging, then it would be consistent in any case. The function does: 1. check if the error is nil or not 2. if not nil, then prints logs 3 and then fail with the error

> If this PR plans to fix https://github.com/intel/intel-device-plugins-for-kubernetes/issues/1252, let's do that for all of the devices then and so that they work the same. Note that the ask for https://github.com/intel/intel-device-plugins-for-kubernetes/issues/1252 to not log output when there are no errors and only log when the sample app fails.

Ok, with the function I made, we can also achieve that only when there are errors, logs are printed.

**Comment by hj-johannes-lee** [comment]
Ready for review!

**Review by tkatila — CHANGES_REQUESTED** [review]
nitpick, can you squash commits into one or two? I don't think each device merits a separate commit.

Otherwise, looks good!

File: test/e2e/utils/utils.go
Comment (hj-johannes-lee) [reply test/e2e/utils/utils.go:48]: This is when we cannot get pod logs. If we get the logs, it will log both logs and the `errPodNotSuccess`.
Do you think it makes more sense to have `framework.Logf("unable to get log from pod: %v", err)`

File: test/e2e/utils/utils.go
Comment (hj-johannes-lee) [reply test/e2e/utils/utils.go:45]: I opened  #1490. Can I do later after this is merged?

File: test/e2e/utils/utils.go
Comment (hj-johannes-lee) [reply test/e2e/utils/utils.go:52]: put in the `logf`

File: test/e2e/utils/utils.go
Comment (mythi) [reply test/e2e/utils/utils.go:45]: What would be the reason to postpone? Usually code review feedback is address *before* merging.

Comment (mythi) [reply test/e2e/utils/utils.go:45]: just this one so that we don't have to change it twice

Comment (hj-johannes-lee) [reply test/e2e/utils/utils.go]: Ah, ok, done!

File: test/e2e/utils/utils.go
Comment (mythi) [inline test/e2e/utils/utils.go:49]: what Gomega prints when the assertion fails? I feel this is redundant.

File: test/e2e/utils/utils.go
Comment (mythi) [inline test/e2e/utils/utils.go:47]: `return fmt.Sprintf()`?

File: test/e2e/utils/utils.go
Comment (hj-johannes-lee) [reply test/e2e/utils/utils.go:50]: so you mean, we do not need errPodNotSuccess at all here, right?

Comment (hj-johannes-lee) [reply test/e2e/utils/utils.go:50]: Then, let me just make it print logs, not error.
From the name you suggested earlier, I thought.. we still print errors inside the function.

Let me change it to `GetPodLogs`

File: test/e2e/utils/utils.go
Comment (hj-johannes-lee) [reply test/e2e/utils/utils.go]: done

**Comment by hj-johannes-lee** [comment]
Idk why, but after changing to gomega way, linter govet (shadow the declaration of err earlier) complains a lot in many random places though there was no change there.

Actually in the first place when I changed the structure of e2e tests to have layers like `Context()` or `BeforeEach()`, we should have considered this matter.

But, anyway, I included in this PR in the first commit to prevent all the possible problems in the future.!

**Review by mythi — APPROVED** [review]

File: test/e2e/qat/qatplugin_dpdk.go
Comment (mythi) [inline test/e2e/qat/qatplugin_dpdk.go:221]: is this missing `gomega.Expect(err).To(gomega.BeNil(), utils.GetPodLogs(...))`

Comment (hj-johannes-lee) [reply test/e2e/qat/qatplugin_dpdk.go]: Ah, yes, thanks.!

**Comment by hj-johannes-lee** [comment]
Something else to be chaged?

**Comment by mythi** [comment]
> Something else to be chaged?

@tkatila needs to approve to unblock merging but e2e-* are also not healthy so we will want to get everything working again before merging

File: test/e2e/dlb/dlb.go
Comment (tkatila) [inline test/e2e/dlb/dlb.go:51]: So there's no logical change, just the names of the variables change?
If that's the case, I think we should revert back to the original form. I feel the explicit error name is not needed. It's clear from the context.

Comment (hj-johannes-lee) [reply test/e2e/dlb/dlb.go:51]: Mikko also asked about this problem.

If you see my commit message, the reason why i changed it to `errFailedToLocateRepoFile` is because of the parts that come after this declaration and still do declaring in the form `X, err := Y`.
This makes linter complains saying it shadows the declaration in the place where I made changes in the commit.

So, we need to choose
1. name the first declaration of 'err' to have another name (that's what I did in this commit)
2. name all the other 'err' to another name.
3. make the linter not complain about this.!

Comment (hj-johannes-lee) [reply test/e2e/dlb/dlb.go:51]: But.... idk why, for some reason... it does not reproduce any more..!
Let me push again without this commit, and see how the ci/cd linter reacts.!

Comment (tkatila) [reply test/e2e/dlb/dlb.go:51]: Ok, I see. Then I'm fine with this version.

Comment (hj-johannes-lee) [reply test/e2e/dlb/dlb.go]: Ah, yeah still it happens.
> test/e2e/sgx/sgx.go:68:12: shadow: declaration of "err" shadows declaration at line 51 (govet)
                podList, err := e2epod.WaitForPodsWithLabelRunningReady(ctx, f.ClientSet, f.Namespace.Name,
                         ^
test/e2e/sgx/sgx.go:85:7: shadow: declaration of "err" shadows declaration at line 51 (govet)
                        if err := utils.WaitForNodesWithResource(ctx, f.ClientSet, "sgx.intel.com/epc", 150*time.Second); err != nil {
                           ^
make: *** [Makefile:87: lint] Error 1

Is it ok if I put the `errFailedToLocateRepoFile` for the first declarations?

Comment (tkatila) [reply test/e2e/dlb/dlb.go]: It fails for you locally? It seems to pass on GH's link.

Comment (hj-johannes-lee) [reply test/e2e/dlb/dlb.go]: Actually the succeeded test contains
`errPodNotRunning` for sgx test.

One other reason why we need to have just another declaration is because of the matter of 'multiple processes' that we have to consider (and still need fixes in many other places).

If we still have 'err' and do not declare there again, it uses the firstly declared one, which means, it may cause some race condition about using err in different processes.

https://github.com/intel/intel-device-plugins-for-kubernetes/pull/1470/files#diff-89c583b2a40dff988d0d71e9e6cd0c4b192c287a11e63afebb7ffc48bfc234b6R85-R93

**Review by tkatila — APPROVED** [review]

File: test/e2e/dlb/dlb.go
Comment (hj-johannes-lee) [reply test/e2e/dlb/dlb.go]: @tkatila Sorry Tuomas, I didn't understand what you meant clearly.
Do we keep it as `err` or is it ok to change them to `errPodNotRunning`?

Comment (tkatila) [reply test/e2e/dlb/dlb.go]: Sorry for the trouble. I didn't understand the context. Please bring the renaming commit back and I'll approve.

**Review by tkatila — DISMISSED** [review]

**Comment by hj-johannes-lee** [comment]
@mythi Could you merge it?

**Comment by hj-johannes-lee** [comment]
@tkatila Sorry, could you approve again? I should have ran `make lint` before pushing.

**Review by tkatila — APPROVED** [review]

**Comment by mythi** [comment]
> @tkatila Sorry, could you approve again? I should have ran `make lint` before pushing. @mythi maybe ready to merge?

I'd propose we wait e2e-fpga back online since this is not urgent

**Review by mythi — APPROVED** [review]

**Pull request merged**
